### DISCUSSION TITLE
Ender 3 pro with direct drive, stepper motor slamming into frame

### DISCUSSION THREAD

**Discussion by Xionite02** [mriscoc/Ender3V2S1#1352] [Q&A]
hi, i was reffered to here by the creality discord.

TLDR: i want a .BIN file for my ender 3 pro with bigtreetech SKR Mini E3 V3.0, all stock components + BLtouch, to have a reduced bed size of 190MM to whenever i do bed levelling it wont slam into the right frame.

right as of now i have a working printer with the Bltouch for auto homing but whenever i try to bed level it does not have enough space
i saw by some online tutorials that it was possible to change the bed size from the default 235mm to 190mm 

so i downloaded the stable version of marlin and the config for the e3 v3, but when i tried to compile the unedited file in vscode with all the correct extensions it will not compile and give alot of undefined errors.
![Schermafbeelding 2024-09-19 185749](https://github.com/user-attachments/assets/2be59778-fb6f-4a17-9f71-aabbcbe8d790)
![Schermafbeelding 2024-09-19 185857](https://github.com/user-attachments/assets/1ab47fee-cffc-476f-87e1-3ed7f26b618f)

all im asking for is:

A: an explanation of why it wont compile (through a discord vc or something like this

B: a working version for the stock ender 3 pro components but with the E3 V3 and bltouch

thank you for your time.

**Comment by mriscoc** [comment]
If you have a DWIN/DACAI/TJC compatible screen, just use the SKR BLT firmware version ([Ender3V2-SKRME3V3-BLTUBL-MPC-20240125.bin](https://github.com/mriscoc/Ender3V2S1/releases/download/20240122/Ender3V2-SKRME3V3-BLTUBL-MPC-20240125.bin)) and follow the calibration guide: https://github.com/mriscoc/Ender3V2S1/wiki/Calibration-Guides

For the old monochrome screen you may try this experimental demo version: [Experimental-DEMO-Ender3-SKRME3V3-BLTUBL-CR10DSPLY-LA-20240508.bin](https://github.com/mriscoc/Special_Configurations/releases/download/demo/Experimental-DEMO-Ender3-SKRME3V3-BLTUBL-CR10DSPLY-LA-20240508.bin)
